src/Observability/Runtime/Common/EnvironmentUtils.cs

```diff
@@ -66,6 +58,40 @@ public static bool IsDevelopmentEnvironment()
             return string.Equals(environment, DevelopmentEnvironmentName, StringComparison.OrdinalIgnoreCase);
         }
 
+        /// <summary>
+        /// Initializes the cached configuration values for environment utilities. Should be called once at application startup.
+        /// </summary>
+        /// <param name="configuration">The configuration instance.</param>
+        /// <param name="force">When true, re-initializes even if already initialized.</param>
+        public static void Initialize(IConfiguration? configuration, bool force = false)
```

Comment:
Instead forcing this to be static and use this `Initialize` method, do you think it would make sense to convert this class to instantiable? Meaning we make the methods non-static and do this `IConfiguration` in the constructor?